src/resolver.rs

```diff
@@ -2315,129 +2316,156 @@ impl<'a> ResolveReport<'a> {
             return Ok(None);
         };
 
-        let mut cache = Cache::acquire(cfg)?;
-        let mut suggestions = vec![];
-        let mut total_lines: u64 = 0;
-        for (&failure_idx, audit_failure) in &fail.failures {
-            let package = &self.graph.nodes[failure_idx];
-            let result = &self.results[failure_idx];
-
-            // Precompute some "notable" parents
-            let notable_parents = {
-                let mut reverse_deps = self.graph.nodes[failure_idx]
-                    .reverse_deps
-                    .iter()
-                    .map(|&parent| self.graph.nodes[parent].name.to_string())
-                    .collect::<Vec<_>>();
-
-                // To keep the display compact, sort by name length and truncate long lists.
-                // We first sort by name because rust defaults to a stable sort and this will
-                // have by-name as the tie breaker.
-                reverse_deps.sort();
-                reverse_deps.sort_by_key(|item| item.len());
-                let cutoff_index = reverse_deps
-                    .iter()
-                    .scan(0, |sum, s| {
-                        *sum += s.len();
-                        Some(*sum)
-                    })
-                    .position(|count| count > 20);
-                let remainder = cutoff_index.map(|i| reverse_deps.len() - i).unwrap_or(0);
-                if remainder > 1 {
-                    reverse_deps.truncate(cutoff_index.unwrap());
-                    reverse_deps.push(format!("and {} others", remainder));
-                }
-                reverse_deps.join(", ")
-            };
+        let cache = Cache::acquire(cfg)?;
+
+        let mut suggestions = tokio::runtime::Handle::current()
+            .block_on(join_all(fail.failures.iter().map(
+                |(failure_idx, audit_failure)| async {
+                    let failure_idx = *failure_idx;
```

Comment:
```suggestion
                |(&failure_idx, audit_failure)| async {
```

no?

Reply:
That unfortunately doesn't work. The async closure apparently ends up capturing a reference with the lifetime of the argument to the function (i.e. `&&'a usize` rather than `&'a usize`) in that situation, and you get a borrowck error. I think it has something to do with `async` captures and patterns.

You can see a super minimal example with this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ed9f8169788bc5de74206bef2907450b

Reply:
bizarre, ok